src/components/admin/form-components/SearchBar.jsx

```diff
@@ -97,23 +127,25 @@ export class SearchBar extends React.Component {
 
 SearchBar.propTypes = {
   handleClick: PropTypes.func.isRequired,
-  suggestions: PropTypes.arrayOf(
-    PropTypes.shape({
-      title: PropTypes.string.isRequired,
-      slug: PropTypes.string,
-      id: PropTypes.number.isRequired,
-    }),
-  ),
-  updateSuggestions: PropTypes.func.isRequired,
+  getSuggestions: PropTypes.func.isRequired,
+  mode: PropTypes.string.isRequired,
   fullWidth: PropTypes.bool,
   disabled: PropTypes.bool,
   enableAdd: PropTypes.bool,
-  category: PropTypes.string.isRequired,
+  slugify(props, propName) {
+    if (
+      props.enableAdd === true &&
+      (props[propName] === undefined || typeof props[propName] !== 'function')
+    ) {
+      return new Error('Please provide a slugify function!');
+    }
+    return null;
+  },
 };
 
 SearchBar.defaultProps = {
-  suggestions: [],
   fullWidth: false,
   disabled: false,
   enableAdd: false,
+  slugify: () => {},
```

Comment:
Should this maybe just be null? Or am I being too picky now? I just wouldn't want it to fail silently if someone accidentally tried accessing `slugify` while `enableAdd` is true

Reply:
Pickiness is great, love it :)!